velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -327,6 +329,84 @@ TpchPlan TpchQueryBuilder::getQ3Plan() const {
   return context;
 }
 
+// select
+// 	o.orderpriority,
+// 	count(*) as order_count
+// from
+// 	orders as o
+// where
+// 	o.orderdate >= '1993-07-01'
+// 	and cast(o.orderdate as date) < date '1993-07-01' + interval '3' month
+// 	and exists (
+// 		select
+// 			*
+// 		from
+// 			lineitem as l
+// 		where
+// 			l.orderkey = o.orderkey
+// 			and l.commitdate < l.receiptdate
+// 	)
+// group by
+// 	o.orderpriority
+// order by
+// 	o.orderpriority;
+
+TpchPlan TpchQueryBuilder::getQ4Plan() const {
+    std::vector<std::string> ordersColumns = {"o_orderdate", "o_orderpriority", "o_orderkey"};
+    std::vector<std::string> lineitemColumns = {"l_orderkey", "l_commitdate", "l_receiptdate"};
+    
+    auto ordersSelectedRowType = getRowType(kOrders, ordersColumns);
+    const auto& ordersFileColumns = getFileColumnNames(kOrders);
+    auto lineitemSelectedRowType = getRowType(kLineitem, lineitemColumns);
+    const auto& lineitemFileColumns = getFileColumnNames(kLineitem);
+    
+    std::string orderDateFilter = formatDateFilter("o_orderdate", ordersSelectedRowType, "'1993-07-01'", "'1993-09-30'");
+    
+    auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+    core::PlanNodeId lineitemPlanNodeId;
+    core::PlanNodeId ordersPlanNodeId;
+
+    auto orders = PlanBuilder(planNodeIdGenerator, pool_.get())
+                        .tableScan(
+                            kOrders,
+                            ordersSelectedRowType,
+                            ordersFileColumns,
+                            {orderDateFilter})
+                        .capturePlanNodeId(ordersPlanNodeId)
+                        .planNode();
+
+    auto plan =
+        PlanBuilder(planNodeIdGenerator, pool_.get())
+            .tableScan(
+                kLineitem,
+                lineitemSelectedRowType,
+                lineitemFileColumns,
+                {},
+                "l_commitdate < l_receiptdate")
+            .capturePlanNodeId(lineitemPlanNodeId)
+            .finalAggregation({"l_orderkey"}, {}, {})
+            // .partialAggregation({"l_orderkey"}, {}, {})
```

Comment:
Why are these lines commented out ?

Reply:
Sorry I read somewhere that a finalAggregation comes after a partialAggregation and so I was adding these in but realized it was not needed and Pramod had it with only finalAggregation from his PR so that got it to work here as well...Will delete these comments and also remove the sql comment on top of the query plan